### ISSUE TITLE
noUnusedParameters always enabled by default when using typescript from node_modules

### ISSUE THREAD

**Issue by Neonit** [TypeStrong/atom-typescript#1433]
## Problem
If Typescript is installed locally in `node_modules`, `atom-typescript` uses it instead of the version shipped with the package. This causes a behavior as if `noUnusedParameters` was enabled. Setting it explicitly to `false` didn't solve the problem for me.

## Reproduction
I was using `atom-typescript` 12.5.3 when reproducing this like so:
1. In an empty folder, open Atom and create a file `foo.ts`:
```
function foo(x: string)
{}
```
2. No errors or warnings should be reported.
3. Now open a console in that folder and install Typescript locally: `npm i typescript`.
4. Restart or reload (Ctrl + Shift + F5) Atom.
5. Once the package has been initialized, the parameter `x` of `foo` should get a blue squiggly underline and the compiler should complain about an unused parameter.
6. Running `tsc foo.ts` should not output any warnings or errors.
(No `tsconfig.json` was used for reproduction)

## Expectation
The expected behavior is that `atom-typescript` just plays by the rules set in the `tsconfig.json`, no matter whether it uses it's own or a locally installed version of Typescript. I haven't tested whether this affects any other settings.

**Comment by lierdakil** [comment]
This seems to be an issue with TypeScript 2.9, which apparently reports those diagnostics regardless of tsconfig settings.

I could not reproduce on TypeScript 2.8.

For additional context, these diagnostics are reported via `suggestionDiag` (which is a new feature in TypeScript v2.8)

If you could open an issue on [Microsoft/TypeScript](https://github.com/Microsoft/TypeScript/issues), that would be very helpful.

The best we could offer from our end is #1434, but it will take some time to polish

(we could of course ignore all `suggestion`-type diagnostics, but that would be throwing the baby out with the bathwater, as far as I'm concerned)

**Comment by Neonit** [comment]
I would open an issue in the Typescript repo, but I lack the background and knowledge to fulfill all their requirements there. I just figured out suggestion diagnostics is a service that provides you in this case with the information that the parameters there are unused and shouldn't be?

They require a test with typescript@next in case the issue is already resolved in the newest version. I don't know how to make atom-typescript use typescript@next nor how to create a suitable minimal example without atom-typescript.

Preparing the issue as desired is out of my time frame, sorry. I'd be happy if someone with the necessary knowledge could help out here as this wouldn't take much of their time.

**Comment by lierdakil** [comment]
Hmm. Went into this rabbit hole, and traced this back to https://github.com/Microsoft/TypeScript/pull/22361 -- so apparently this is very deliberate.

**Comment by lierdakil** [comment]
Okay, so there is an "escape hatch" so to speak. I've added necessary checks to https://github.com/TypeStrong/atom-typescript/pull/1434. Will try to cook up a release this weak.

In the meantime, consider using TypeScript 2.8 if you can.

For the record, atom-typescript uses whatever TypeScript version you have installed into `node_modules`. If there isn't one, only then it will fallback to the bundled version, which happens to be TypeScript 2.8 at the moment.

**Comment by Neonit** [comment]
Thanks for your engagement. Take your time, though. It's a bit annoying, but it doesn't hinder development.

**Comment by lierdakil** [comment]
v12.6.0 is now released :tada: which includes the following setting:
![image](https://user-images.githubusercontent.com/7275622/41909505-d949476a-794f-11e8-8f6b-727c4a635f58.png)
(disabled by default for now, because I assume TypeScript team knows what they're doing, but it might be enabled by default later in a patch-release if more people complain)

Hopefully this is satisfactory.
